ngraph/core/reference/src/runtime/reference/fft.cpp

```diff
@@ -0,0 +1,594 @@
+                    }
+
+                    return input_data[src_index + offset];
+                }
+
+                void copy_data_from_input(complex_type* result,
+                                          const complex_type* input_data,
+                                          int64_t src_index,
+                                          int64_t fft_size,
+                                          const std::vector<int64_t>& fft_strides,
+                                          const std::vector<int64_t>& input_fft_lengths,
+                                          const std::vector<int64_t>& input_fft_strides)
+                {
+                    for (int64_t idx = 0; idx < fft_size; ++idx)
+                    {
+                        auto coords = coords_from_index(idx, fft_strides);
+                        complex_type value = get_value_from_input(
+                            input_data, src_index, coords, input_fft_lengths, input_fft_strides);
+                        result[idx] = value;
+                    }
+                }
+
+                bool blob_is_zero(const complex_type* data, int64_t blob_size)
+                {
+                    for (int64_t i = 0; i < blob_size; ++i)
+                    {
+                        if (data[i] != complex_type{0.0f, 0.0f})
+                        {
+                            return false;
+                        }
+                    }
+                    return true;
+                }
+
+                int64_t offset_from_coords_and_strides(const std::vector<int64_t>& coords,
+                                                       const std::vector<int64_t>& strides)
+                {
+                    int64_t offset = 0;
+                    int64_t num_of_axes = coords.size();
+                    for (int64_t i = 0; i < num_of_axes; ++i)
+                    {
+                        offset += coords[i] * strides[i];
+                    }
+                    return offset;
+                }
+
+                void copy_data_to_output(complex_type* output,
+                                         const complex_type* data,
+                                         int64_t dst_index,
+                                         int64_t fft_size,
+                                         const std::vector<int64_t>& fft_strides,
+                                         const std::vector<int64_t>& output_fft_strides)
+                {
+                    for (int64_t idx = 0; idx < fft_size; ++idx)
+                    {
+                        auto coords = coords_from_index(idx, fft_strides);
+                        complex_type value = data[idx];
+                        int64_t offset = offset_from_coords_and_strides(coords, output_fft_strides);
+
+                        output[dst_index + offset] = value;
+                    }
+                }
+
+                static constexpr float pi = 3.141592653589793238462643f;
```

Comment:
If I remember right the `cmath` library contains `M_PI` constant.

Reply:
It is not true: see https://en.cppreference.com/w/cpp/header/cmath